### PR TITLE
Ability to set server PORT number via keystone config

### PR DESCRIPTION
**Pull request by MurzNN**
Comparing with #4668 - yere is another (guess better) way to fix problem with configuring server PORT number from config or `.env` file.
I move PORT assigning process from `util.ts` to start scripts, also extract config parsing from async to upper process. Hope, this is better way.

### CODE REVIEW

File: packages-next/types/src/core.ts [hunk 1]
```diff
@@ -92,6 +92,7 @@ export type KeystoneConfig = {
   server?: {
     /** Configuration options for the cors middleware. Set to true to core Keystone defaults */
     cors?: any;
+    port?: number;
```
Comment (timleslie) [inline packages-next/types/src/core.ts:95]:
```suggestion
    /** Port number to start the server on. Defaults to process.env.PORT || 3000 */
    port?: number;
```

File: packages-next/keystone/src/scripts/run/dev.ts [hunk 2]
```diff
@@ -58,6 +58,7 @@ export const dev = async ({ dotKeystonePath, projectAdminPath }: StaticPaths, sc
     if (expressServer) return expressServer(req, res, next);
     res.sendFile(devLoadingHTMLFilepath);
   });
+  const PORT = config.server?.port || process.env.PORT || 3000;
   server.listen(PORT, (err?: any) => {
```
Comment (timleslie) [inline packages-next/keystone/src/scripts/run/dev.ts:62]:
```suggestion
  const port = config.server?.port || process.env.PORT || 3000;
  server.listen(port, (err?: any) => {
```

File: packages-next/keystone/src/scripts/run/start.ts [hunk 3]
```diff
@@ -29,6 +28,7 @@ export const start = async ({ dotKeystonePath, projectAdminPath }: StaticPaths)
   );
   console.log(`👋 Admin UI and graphQL API ready`);
 
+  const PORT = config.server?.port || process.env.PORT || 3000;
   server.listen(PORT, (err?: any) => {
```
Comment (timleslie) [inline packages-next/keystone/src/scripts/run/start.ts:32]:
```suggestion
  const port = config.server?.port || process.env.PORT || 3000;
  server.listen(port, (err?: any) => {
```

**Review by timleslie — CHANGES_REQUESTED** [review]
Thanks for this PR @MurzNN, I think this is the right way to go about setting the port. Just a few minor changes required. Also, could you please add a changeset with a `minor` bump to the `keystone-next/keystone` package. See the changesets bot message for instructions on how to do this (tl;dr run `yarn changeset` and follow instructions). Thanks!

**Comment by MurzNN** [comment]
@timleslie Thanks for reviewing, I fixed all suggestions!

File: packages-next/types/src/core.ts [hunk 4]
```diff
@@ -91,8 +91,10 @@ export type KeystoneConfig = {
   session?: () => SessionStrategy<any>;
   ui?: KeystoneAdminUIConfig;
   server?: {
-    /** Configuration options for the cors middleware. Set to `true` to use core Keystone defaults */
+    /** Configuration options for the cors middleware. Set to true to core Keystone defaults */
```
Comment (timleslie) [inline packages-next/types/src/core.ts:94]:
```suggestion
    /** Configuration options for the cors middleware. Set to `true` to use core Keystone defaults */
```

File: .changeset/loud-singers-check.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+'@keystone-next/keystone': minor
+'@keystone-next/types': minor
+---
+
+Ability to set server port number via keystone config
```
Comment (timleslie) [inline .changeset/loud-singers-check.md:6]:
```suggestion
Added the ability to set the server port number via `config.server.port`.
```

File: packages-next/keystone/src/scripts/run/dev.ts [hunk 6]
```diff
@@ -58,7 +58,8 @@ export const dev = async ({ dotKeystonePath, projectAdminPath }: StaticPaths, sc
     if (expressServer) return expressServer(req, res, next);
     res.sendFile(devLoadingHTMLFilepath);
   });
-  server.listen(PORT, (err?: any) => {
+  const port = config.server?.port || process.env.PORT || 3000;
+  server.listen(port, (err?: any) => {
     if (err) throw err;
     console.log(`⭐️ Dev Server Ready on http://localhost:${PORT}`);
```
Comment (timleslie) [inline packages-next/keystone/src/scripts/run/dev.ts:64]:
```suggestion
    console.log(`⭐️ Dev Server Ready on http://localhost:${port}`);
```

File: packages-next/keystone/src/scripts/run/start.ts [hunk 7]
```diff
@@ -29,7 +28,8 @@ export const start = async ({ dotKeystonePath, projectAdminPath }: StaticPaths)
   );
   console.log(`👋 Admin UI and graphQL API ready`);
 
-  server.listen(PORT, (err?: any) => {
+  const port = config.server?.port || process.env.PORT || 3000;
+  server.listen(port, (err?: any) => {
     if (err) throw err;
     console.log(`⭐️ Server Ready on http://localhost:${PORT}`);
```
Comment (timleslie) [inline packages-next/keystone/src/scripts/run/start.ts:34]:
```suggestion
    console.log(`⭐️ Server Ready on http://localhost:${port}`);
```

**Review by timleslie — CHANGES_REQUESTED** [review]

**Review by timleslie — APPROVED** [review]
Looks good, thanks for the contribution!

**Pull request merged**